src/common/state/local_scheduler_table.h

```diff
@@ -0,0 +1,65 @@
+#ifndef LOCAL_SCHEDULER_TABLE_H
+#define LOCAL_SCHEDULER_TABLE_H
+
+#include "db.h"
+#include "table.h"
+
+typedef struct {
+  int task_queue_length;
+  int available_workers;
```

Comment:
I would throw in total_workers as well, but we can make extensions a separate PR.
This is the nice thing about putting everything in a struct 👍 . Easy to extend.